### DISCUSSION TITLE
how to set snowflake query_tag for ibis snowflake backend.

### DISCUSSION THREAD

**Discussion by ruiyang2015** [ibis-project/ibis#7998] [Q&A] [answered]
Search the docs and also tried using chatgpt, none give a good answer on how to set the connection and session level query_tag for ibis.snowflake backend, please help

**Comment by lostmygithubaccount** [comment]
hi @ruiyang2015, is that something that can be set in the Snowflake Python connector? Ibis passes through additional kwargs to the underlying connection library if so

**Comment by cpcloud** [comment, marked as answer]
You can set this using

```python
con.raw_sql("ALTER SESSION SET QUERY_TAG = 'my_tag'")
```

or by passing the appropriate `session_parameters` argument when connecting:

```python
ibis.snowflake.connect(
    ...,
    session_parameters={"QUERY_TAG": "my_tag"}
)
```
